localstack/services/apigateway/invocations.py

```diff
@@ -192,18 +176,19 @@ def is_api_key_valid(invocation_context: ApiInvocationContext) -> bool:
     ).apigateway
     rest_api = client.get_rest_api(restApiId=invocation_context.api_id)
 
-    if rest_api.get("apiKeySource") != "HEADER":
-        # When the apiKeySource is set to AUTHORIZER, the authorizer is supposed to return the API key as a field
-        # `usageIdentifierKey`
-        # see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html
-        # Authorizers are only mocked, so we can't validate the key. Return True in that case
-        return True
-
-    api_key = invocation_context.headers.get("X-API-Key")
-    if not api_key:
-        return False
+    # The source of the API key for metering requests according to a usage plan.
+    # Valid values are:
+    # - HEADER to read the API key from the X-API-Key header of a request.
+    # - AUTHORIZER to read the API key from the UsageIdentifierKey from a custom authorizer.
 
-    return validate_api_key(api_key, invocation_context)
+    api_key_source = rest_api.get("apiKeySource")
+    match api_key_source:
+        case "HEADER":
+            api_key = invocation_context.headers.get("X-API-Key")
+            return validate_api_key(api_key, invocation_context) if api_key else False
+        case "AUTHORIZER":
+            api_key = invocation_context.auth_identity.get("apiKey")
```

Comment:
this is populated on the authorizer logic upstream, if `usageIdentifierKey` is populated.

Reply:
I know this is annoying, but would there be a way to short circuit if the authorizer hasn't run? Some way to know that this is only community, so that we don't block people who don't have access to authorizers, and return True in that case? 😬 sorry for piling up

Reply:
Well the way I had before was sensible to that (minus the return that could be easily changed), because either was implemented or not. Now moving the logic out of authorizers and `auth_identity` being the place where that is populated if the authorizer is executed, blocked if not. I don't think we have a way, proper way to check on features are enabled of disabled and definitely not want to hack around without proper checkpoints

Reply:
Okay, then let's see. People can also remove the authorizer from their setup if this is blocking.

Reply:
patches bring their own issues with debugging and refactoring since moving code around implies changes on `-ext` that are invisible until that code is executed. There is definitely a balance I'm just on the opinion that the patch approach is not developer friendly

Reply:
Agreed, I'm not fond of them either. I think we'll go forward with that!